doc/src/modules/combinatorics/fp_groups.rst

```diff
@@ -0,0 +1,290 @@
+
+* ``n``: A non-negative integer, non-mutable attriburte, dependently
+  calculated as the maximum among the live-cosets (i.e `\Omega`).
+
+* ``table``: A list of lists, mutable attribute, mathematically represents the
+  coset table.
+
+* ``omega``: A list, dependent on the internal attribute ``p``. `\Omega`
+  represents the list of live-cosets. A *standard* coset-table has its
+  `\Omega = [0, 1, \ldots, index-1]` where `index` is the index of subgroup
+  ``H`` in ``G``.
+
+For experienced users we have a number of parameters that can be used to
+manipulate the algorithm, like
+
+* ``coset_table_max_limit`` (default value = `4096000`): manipulate the maximum
+  number of cosets allowed in coset enumeration, i.e the number of rows allowed
+  in coset table. A coset enumeration will not finish if the subgroup does not
+  have finite index, and even if it has it may take many more intermediate
+  cosets than the actual index of the subgroup is. To avoid a coset enumeration
+  "running away" therefore SymPy has a "safety stop" built-in. This is
+  controlled by this variable. For example:
+
+  >>> CosetTable.coset_table_max_limit = 50
+  >>> F, a, b = free_group("a, b")
+  >>> Cox = FpGroup(F, [a**6, b**6, (a*b)**2, (a**2*b**2)**2, (a**3*b**3)**5])
+  >>> C_r = coset_enumeration_r(Cox, [a])
+  Traceback (most recent call last):
+    ...
+  ValueError: the coset enumeration has defined more than 50 cosets
+
+
+* ``max_stack_size`` (default value = `500`): manipulate the maximum size of
+  ``deduction_stack`` above or equal to which the stack is emptied.
+
+Compression and Standardization
+```````````````````````````````
+
+For any two entries `i, j` with `i < j` in coset table, the first
+occurrence of `i` in a coset table precedes the first occurrence of `j` with
+respect to the usual row-wise ordering of the table entries. We call such a
+table a standard coset table. To standardize a ``CosetTable`` we use the
+``.standardize`` method.
+
+**Note** the method alters the given table, it does not create a copy.
+
+Subgroups of Finite Index
+-------------------------
+
+The functionality in this section are concerned with the construction of
+subgroups of finite index. We describe a method for computing all subgroups
+whose index does not exceed some (modest) integer bound.
+
+Low Index Subgroups
+```````````````````
+
+``low_index_subgroups(G, N)``: Given a finitely presented group `G = <X \mid R>`
+(can be a free group), and ``N`` a positive integer, determine the conjugacy classes of
+subgroups of ``G`` whose indices is less than or equal to ``N``.
+
+For example to find all subgroups of `G = <a, b \mid a^2 = b^3 = (ab)^4 = 1>`
+having index <= 4, can be found as follows:
+
+>>> from sympy.combinatorics.fp_groups import low_index_subgroups
+>>> F, a, b = free_group("a, b")
+>>> G = FpGroup(F, [a**2, b**3, (a*b)**4])
+>>> l = low_index_subgroups(G, 4)
+>>> for coset_table in l:
+...     print(coset_table.table)
```

Comment:
I get an error currently
```
gaurav@stallman:~/Public/sympy$ python3 ./bin/doctest doc/src/modules/combinatorics/fp_groups.rst 
**********************************************************************
File "/home/gaurav/Public/sympy/doc/src/modules/combinatorics/fp_groups.rst", line 238, in fp_groups.rst
Failed example:
    for coset_table in l:
        print(coset_table.table)
Expected nothing
Got:
    [[0, 0, 0, 0]]
    [[0, 0, 1, 2], [1, 1, 2, 0], [3, 3, 0, 1], [2, 2, 3, 3]]
    [[0, 0, 1, 2], [2, 2, 2, 0], [1, 1, 0, 1]]
    [[1, 1, 0, 0], [0, 0, 1, 1]]
**********************************************************************
1 items had failures:
   1 of  24 in fp_groups.rst
***Test Failed*** 1 failures.
========================================================================== rst doctests start ===========================================================================
executable:         /usr/bin/python3  (3.4.3-final-0) [CPython]
architecture:       64-bit
cache:              yes
ground types:       python 
hash randomization: on (PYTHONHASHSEED=2510432324)

doc/src/modules/combinatorics/fp_groups.rst [24]                                                                                                                      [1]

DO *NOT* COMMIT!
```
Do you know what is the problem with this? I expected `print` statement to work fine.